src/test/java/org/broadinstitute/consent/http/service/dao/DraftServiceDAOTest.java

```diff
@@ -0,0 +1,242 @@
+    when(gcsService.storeDocument(any(), anyString(), any())).thenReturn(
+        BlobId.of(UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+    User user = createUser();
+    DraftInterface draft = createDraft(user, 3);
+    Map<String, FormDataBodyPart> files = getRandomFiles(1);
+    List<FileStorageObject> addedAttachments = draftServiceDAO.addAttachments(draft, user, files);
+    assertThat(addedAttachments, hasSize(1));
+    DraftInterface updatedDraft = draftServiceDAO.getAuthorizedDraft(draft.getUUID(), user);
+    assertThat(updatedDraft.getStoredFiles(), hasSize(4));
+  }
+
+  @Test
+  void testDeleteAttachmentFromDraft() throws SQLException, IOException {
+    when(gcsService.storeDocument(any(), anyString(), any())).thenReturn(
+        BlobId.of(UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+    User user = createUser();
+    DraftInterface draft = createDraft(user, 3);
+    Set<FileStorageObject> storedFiles = draft.getStoredFiles();
+    assertThat(storedFiles, hasSize(3));
+    for (FileStorageObject file : storedFiles) {
+      draftServiceDAO.deleteDraftAttachment(draft, user, file.getFileStorageObjectId());
+    }
+    assertThat(draftServiceDAO.getAuthorizedDraft(draft.getUUID(), user).getStoredFiles(),
+        hasSize(0));
+  }
+
+
+
+  @Test
+  void testUpdateDraft() throws SQLException {
+    User user = createUser();
+    DraftInterface draft = createDraft(user, 1);
+    String updatedJson = "{\"study\": \"My example study\"}";
+    String newDraftName = "My favorite draft";
+    String originalDocumentJson = draft.getJson();
+    Date originalDocumentDate = draft.getUpdateDate();
+    assertEquals(originalDocumentJson, draft.getJson());
+    assertEquals(originalDocumentDate, draft.getUpdateDate());
+    assertNotEquals(draft.getName(), newDraftName);
+    draft.setName(newDraftName);
+    draft.setJson(updatedJson);
+    DraftInterface updatedDraft = draftServiceDAO.updateDraft(draft, user);
+    assertEquals(draft.getUUID(), updatedDraft.getUUID());
+    assertEquals(newDraftName, updatedDraft.getName());
+    assertEquals(updatedJson, updatedDraft.getJson());
+    assertThinUser(updatedDraft.getCreateUser());
+    assertThinUser(updatedDraft.getUpdateUser());
+  }
+
+  @NotNull
+  private DraftInterface createDraft(User user, Integer numberOfFiles)
+      throws SQLException {
+    DraftStudyDataset draft = new DraftStudyDataset("{}", user);
+    draftServiceDAO.insertDraft(draft);
+    Map<String, FormDataBodyPart> mapOfFiles = getRandomFiles(numberOfFiles);
+    draftServiceDAO.addAttachments(draft, user, mapOfFiles);
+    return draftServiceDAO.getAuthorizedDraft(draft.getUUID(), user);
+  }
+
+  private void assertThinUser(User user) {
+    assertThat(user.getRoles(), is(nullValue()));
+    assertThat(user.getUserId(), is(notNullValue()));
+    assertThat(user.getInstitutionId(), is(nullValue()));
+    assertThat(user.getEraCommonsId(), is(nullValue()));
```

Comment:
`is()`  is unnecessary here. But maybe it does make it "read" better?
```suggestion
    assertThat(user.getRoles(), nullValue());
    assertThat(user.getUserId(), notNullValue());
    assertThat(user.getInstitutionId(), nullValue());
    assertThat(user.getEraCommonsId(), nullValue());
```